src/odemis/driver/nfpm.py

```diff
@@ -91,8 +92,9 @@ def __init__(self, name, role, address, axes, stepsize, sn=None, **kwargs):
                 raise ValueError("stepsize should be in meter, but got %g" % (sz,))
         self._stepsize = stepsize
 
-        self._address = address
-        self._sn = sn
+        # Convert python2 byte str to unicode str
+        self._address = address.decode() if isinstance(address, bytes) else address
+        self._sn = sn.decode() if isinstance(sn, bytes) else sn
```

Comment:
Why would the user pass these parameters as bytes? I think you can just assume these are strings. Or are there specific cases you want to handle?

Reply:
I thought it would be best if the class also worked with python2 strings (=bytes) as arguments. Is this unnecessary?

Reply:
On Python2, the user passes a string, and it should stay a string, to be used later as an IP address, and as comparison. It works fine for now, no need to make the code more complex by converting the strings to unicode.

Reply:
ok